Milestone 2

bruna-genz opened the pull request:
Hello, please take a look around, in this pull request we:

- Created components, containers, actions, and reducers directories.

- Created actions and reducers to manage the books' state.

- Initialized the Redux store with an initial state in src/index.js.

- Created the UI structure with the components: App, BooksList, BooksForm, and Book.

Currently, the books in the Redux store are being displayed on the browser, as well as the form to add more books. 
You can find the live demo here: https://booky-redux.herokuapp.com/

Gently review our project,
Thanks sincerely,
@Luzaks
@bruna-genz

bruna-genz committed:
Feat: Set up the project structure by creating the actions, components, containers, reducers, styles, and tests directories

bruna-genz committed:
Feat: define BooksList and BooksForm containers and render them on App component

Luzaks committed:
Feat: Initialize the redux store.

Luzaks committed:
Feat: Implement books actions and books reducer.

Luzaks committed:
Feat: Implement the Book component linked to the BookList, connect the BookList to the store.

bruna-genz committed:
Fix typos and linter errors

bruna-genz committed:
Merge pull request #5 from Luzaks/development

Merge 'refreshed' development into Milestone_2

Redvanisation commented on src/containers/BooksList.js:
```diff
@@ -0,0 +1,22 @@
+  </table>
+);
+
+const connectedBookList = connect(mapStateToProps)(BooksList);
```
This can be done in line 22 like this `export default connect(mapStateToProps)(connectedBookList)`

Redvanisation commented on src/reducers/books.js:
```diff
@@ -0,0 +1,12 @@
+    case 'CREATE_BOOK':
+      return { books: [...state, action.payload] };
+    case 'REMOVE_BOOK':
+      return { books: [...state, action.payload] };
```
This will add a new book instead of deleting it, change that to use the `filter` method filtering the state without the book you want to remove.

Redvanisation commented on src/actions/index.js:
```diff
@@ -0,0 +1,26 @@
+  return {
+    type: 'REMOVE_BOOK',
+    payload: {
+      id: book.id,
+      title: book.title,
+      category: book.category,
```
You could pass only the `book.id` here, you don't need the title and the category.

Redvanisation reviewed (changes requested):
# Changes required  ♻️ 

### Hello team  🌟

**Nice work on this project, but the following changes need to be done before moving on ⬇️** 

- **index.js** in `src/reducers` should combine the app's specific reducers.

- **book.js** reducer is not well implemented, please check all my comments below.

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

bruna-genz committed:
Fix: clean code

bruna-genz committed:
Fix: change rootReducer to fix bugs with the initialState

bruna-genz commented:
Hi @Redvanisation,

We implemented the required changes, using combineReducers in src/reducers/index.js, and fixed the Books reducer removeBook action to delete the specific book using the filter method.

Thanks sincerely, 
@Luzaks
@bruna-genz

shloch reviewed (approved):
![accomplished](https://user-images.githubusercontent.com/40712459/86941116-78138d80-c13b-11ea-9b5e-188b2948c6c0.jpg)

### Good to go..........modifications made  :)

Luzaks commented:
@shloch @Redvanisation Thank you so much for your reviews!
